### PR TITLE
Adding source code to produce the `ar/` version of this dataset.

### PR DESCRIPTION
**Pull request by alxmrs**

### CODE REVIEW

File: src/arco_era5/source_data.py [hunk 1]
```diff
@@ -2,20 +2,20 @@
...
 
+import pathlib
 import xarray
 
-import pathlib
-import fsspec
+import numpy as np
+import pandas as pd
+import typing as t
+import xarray as xa
```
Comment (alxmrs) [inline src/arco_era5/source_data.py:15]: Typically, this is imported as `xr`.

File: src/arco_era5/source_data.py [hunk 2]
```diff
@@ -28,7 +28,7 @@
     "date-variable-pressure_level/{year}/{month:02d}/"
     "{day:02d}/{variable}/{pressure_level}.nc")
 
-STATIC_VARIABLES = (
+STATIC_VARIABLES = [
```
Comment (alxmrs) [inline src/arco_era5/source_data.py:31]: Any reason why tuples didn't work?

File: src/arco_era5/source_data.py [hunk 3]
```diff
@@ -213,3 +214,63 @@ def get_var_attrs_dict(root_path=GCP_DIRECTORY):
...
+def daily_date_iterator(start_date: str, end_date: str
+                        ) -> t.Iterable[t.Tuple[int, int, int]]:
+    """Iterates all (year, month, day) tuples between start_date and end_date."""
+    first_day = pd.Timestamp(start_date)
+    final_day = pd.Timestamp(end_date)  # non-inclusive
+    time_delta = pd.Timedelta("1 day")
+    current_day = first_day
+    while current_day < final_day:
+        yield current_day.year, current_day.month, current_day.day
+        current_day += time_delta
```
Comment (alxmrs) [inline src/arco_era5/source_data.py:228]: Can we use `pd.date_range`?

File: src/netcdf_to_zarr.py [hunk 4]
```diff
@@ -18,8 +18,9 @@
       --output_path="gs://gcp-public-data-arco-era5/ar/$USER-1959-2022-full_37-1h-0p25deg-chunk-1.zarr-v2" \
       --pressure_levels_group="full_37" \
       --time_chunk_size=1 \
+      --start_date '1997-01-01' \
+      --end_date '1997-01-02' \
```
Comment (alxmrs) [inline src/netcdf_to_zarr.py:22]: Let's be sure to include the full data date range.

File: setup.py [hunk 5]
```diff
@@ -88,7 +88,7 @@ def run(self):
     platforms=['darwin', 'linux'],
     python_requires='>3.8, <3.11',
     install_requires=[
-        'apache_beam[gcp]',
+        'apache_beam[gcp]==2.40.0',
         'pangeo-forge-recipes==0.9.1',
         'pandas',
         'gcsfs',
```
Comment (alxmrs) [inline setup.py:94]: If we never use `absl-py`, let's remove the dep (below).

File: src/netcdf_to_zarr.py [hunk 6]
```diff
@@ -338,37 +279,44 @@ def define_pipeline(
...
-        raise ValueError(f"time_chunk_size{TIME_CHUNK_SIZE.value} must "
-                         f"evenly divide {num_steps_per_day}")
+    num_steps_per_day = _HOURS_PER_DAY // TIME_RESOLUTION_HOURS
+    if num_steps_per_day % known_args.time_chunk_size != 0:
+        raise ValueError(
+            f"time_chunk_size {known_args.time_chunk_size} must evenly divide {num_steps_per_day}"
+        )
 
-    with beam.Pipeline(runner=RUNNER.value, argv=argv) as root:
+    with beam.Pipeline(options=PipelineOptions(pipeline_args)) as root:
```
[the remark this answers is not in the document]
Comment (alxmrs) [reply src/netcdf_to_zarr.py:309]: (optional).

File: src/arco_era5/source_data.py (hunk 2 above)
Comment (dabhicusp) [reply src/arco_era5/source_data.py]: It's working with tuples,too. Reverted changes from list to tuple. Thanks for the suggestions.

**Comment by alxmrs** [comment]
LGTM. However, I can't approve since I'm a co-author of this CL. We'll have to ask @DarshanSP19 :)

**Review by DarshanSP19 — APPROVED** [review]

**Pull request merged**